assets/js/modules/adsense/components/module/ModuleTopEarningPagesWidget.js

```diff
@@ -0,0 +1,208 @@
+import Link from '../../../../components/Link';
+import PreviewTable from '../../../../components/PreviewTable';
+import TableOverflowContainer from '../../../../components/TableOverflowContainer';
+import AdSenseLinkCTA from '../../../analytics/components/common/AdSenseLinkCTA';
+import { STORE_NAME, DATE_RANGE_OFFSET } from '../../datastore/constants';
+import { CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { MODULES_ANALYTICS } from '../../../analytics/datastore/constants';
+import { isZeroReport } from '../../../analytics/util/is-zero-report';
+import { isRestrictedMetricsError } from '../../../analytics/util/error';
+import { generateDateRangeArgs } from '../../../analytics/util/report-date-range-args';
+import { getCurrentDateRangeDayCount } from '../../../../util/date-range';
+import ModuleHeader from '../common/ModuleHeader';
+const { useSelect } = Data;
+
+function ModuleTopEarningPagesWidget( { Widget, WidgetReportZero, WidgetReportError } ) {
+	const {
+		isAdSenseLinked,
+		data,
+		isLoading,
+		error,
+		analyticsMainURL,
+		currentDayCount,
+	} = useSelect( ( select ) => {
+		const analyticsStore = select( MODULES_ANALYTICS );
+		const userStore = select( CORE_USER );
+
+		const dateRange = userStore.getDateRange();
+		const { startDate, endDate } = userStore.getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+		} );
+
+		const reportArgs = {
+			startDate,
+			endDate,
+			dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
+			metrics: [
+				{ expression: 'ga:adsenseRevenue', alias: 'Earnings' },
+				{ expression: 'ga:adsenseECPM', alias: 'Page RPM' },
+				{ expression: 'ga:adsensePageImpressions', alias: 'Impressions' },
+			],
+			orderby: {
+				fieldName: 'ga:adsenseRevenue',
+				sortOrder: 'DESCENDING',
+			},
+			limit: 10,
+		};
+
+		return {
+			isAdSenseLinked: analyticsStore.getAdsenseLinked(),
+			data: analyticsStore.getReport( reportArgs ),
+			error: analyticsStore.getErrorForSelector( 'getReport', [ reportArgs ] ),
+			isLoading: ! analyticsStore.hasFinishedResolution( 'getReport', [ reportArgs ] ),
+			analyticsMainURL: analyticsStore.getServiceReportURL(
+				'content-publisher-overview',
+				generateDateRangeArgs( { startDate, endDate } )
+			),
+			currentDayCount: getCurrentDateRangeDayCount( dateRange ),
+		};
+	} );
+
+	// A restricted metrics error will cause this value to change in the resolver
+	// so this check should happen before an error, which is only relevant if they are linked.
+	if ( ! isAdSenseLinked ) {
+		return <AdSenseLinkCTA />;
+	}
+
+	if ( error && ! isRestrictedMetricsError( error ) ) {
+		return <WidgetReportError error={ error } moduleSlug="adsense" />;
+	}
+
+	if ( isLoading ) {
+		return <PreviewTable padding />;
+	}
+
+	if ( isZeroReport( data ) ) {
+		return <WidgetReportZero moduleSlug="adsense" />;
+	}
+
+	const { rows } = data?.[ 0 ]?.data || {};
+
+	if ( ! Array.isArray( rows ) ) {
+		return null;
+	}
```

Comment:
Let's display zero component when it's restricted metrics error instead of returning null:

```suggestion
	if ( isZeroReport( data ) || isRestrictedMetricsError( error ) ) {
		return <WidgetReportZero moduleSlug="adsense" />;
	}
```